### DISCUSSION TITLE
How to Compute Metric iou and mean iou

### DISCUSSION THREAD

**Discussion by minwang-ai** [Lightning-AI/torchmetrics#628] [General]
Hi,

The prediction and target shape of  iou class in torch metrics are 

preds (float or long tensor): (N, ...) or (N, C, ...) where C is the number of classes and N is the number of images(batch size)
target (long tensor): (N, ...)

What about other dimension after N .. ?
And what if there is no dimension C as num_class for preds?

How can I compute iou and miou during training and testing, when my training batch size is really small (N=3 ) and the validation dataloader loads data of one image each time (N=1) ?

If I use IOU metric call directly, I only get iou for this batch (miou_step), but I want to average over batches to get mean iou for different class (miou_epoch). What should I do?
